LoC

OkThought opened the pull request:
Implement working task solution

OkThought committed:
Add .idea and LoC/config to ignore list

OkThought committed:
Implement working task solution

OkThought committed:
Implement an Apache Ant build.xml script

OkThought committed:
Add package 'bogush' and move all src files to it

OkThought committed:
Add *.iml

OkThought committed:
Fix, Refactor, Improve

OkThought committed:
Add tests

OkThought committed:
Add throw exception for zero filters in constructor

OkThought committed:
Add getters

OkThought committed:
Surround statistics collector construction with try/catch

OkThought committed:
Fix to count 0 on empty files and (amount of line-breaks + 1) on non-empty files

OkThought committed:
Small changes and fixes
Move FilterParseException class into FilterParser source file
Add throw FilterParseException when no filters parsed
Add skip empty lines

OkThought committed:
Up version

OkThought committed:
Modify gitignore
Add Junit necessary jars (unignore)

OkThought committed:
Add Junit necessary jars

OkThought committed:
modify gitignore

OkThought committed:
Rename bogush package to loc (lines of code)

OkThought committed:
Change package 'bogush' to 'loc'

OkThought committed:
Change package 'bogush' to 'loc'

OkThought committed:
Fix and rewrite

OkThought committed:
Change package hierarchy

OkThought committed:
Rename BufferedPrinter to FilterSequenceStringBuffer

OkThought committed:
Hotfix
Remove OutputStream reference
Fix: pass filter argument to serialize()
Override toString()

OkThought committed:
Rename FilterParser to Parser
Rename FilterParseException to ParseException

OkThought committed:
Rename IFilter to Filter

OkThought committed:
Workaround with Exceptions (Temporal)

OkThought committed:
Rename IFilterSerializer to FilterSerializer

OkThought committed:
Change input files for unit tests
Apply to the new standard

OkThought committed:
Add Serializer to AggregateFilter
Use StringBuffer instead of StringBuilder

OkThought committed:
Rename FilterAnd to And
Fix method serialize():
- pass Filter argument
- implement it's body.
Change IFilter to Filter

OkThought committed:
Rename FilterNot to Not
Implement serialize() in Not.Serializer

OkThought committed:
Rename EarlierModified to ModifiedEarlier
Implement ModifiedEarlier.Serializer.serialize()
Refactor

OkThought committed:
Rename LaterModified to ModifiedLater
Implement ModifiedLater.Serializer.serialize()
Refactor

OkThought committed:
Rename ModifiedTimeFilter to TimeModifiedFilter

OkThought committed:
Implement FileExtensionFilter.Serializer.serialize()

OkThought committed:
Refactor

OkThought committed:
Fix wtf

OkThought committed:
Workaround with Exceptions (Temporal)

OkThought committed:
Fix serialization
Refactor

OkThought committed:
Refactor
Fix indents

OkThought committed:
Fix indents

OkThought committed:
Fix indents

OkThought committed:
Import Filters

OkThought committed:
Use StringBuilder instead of StringBuffer
It works more efficiently on one thread.

OkThought committed:
Rename parse(String) to serialize(String)

OkThought committed:
Fix exceptions. Add throw wrappings to FilterSerializeException or Parser.ParseException.

OkThought committed:
Make filters final

OkThought committed:
Update README.md

OkThought committed:
Create README.md

OkThought committed:
Update README.md

OkThought committed:
Update README.md

OkThought committed:
Remove Parser and Implement FilterStringStream instead

OkThought committed:
Implement FilterSequenceParser

OkThought committed:
Remove try-catch block and use throws instead

OkThought committed:
Rename getters

OkThought committed:
Fix file extension filter creation

OkThought committed:
Remove old input files and re-implement custom test

OkThought committed:
Merge remote-tracking branch 'origin/LoC' into LoC

OkThought committed:
Fix file extensions

OkThought committed:
Fix file path

OkThought committed:
Fix Aggregate.Serializer.serialize(Filter): add prefix

OkThought committed:
Fix Statistics serialization
Add automatic indentation

OkThought committed:
Change exceptions hierarchy
Add new exceptions

levanovd commented on LoC/src/loc/statistics/StatisticsComparator.java:
```diff
@@ -0,0 +1,17 @@
+
+public class StatisticsComparator implements Comparator<Map.Entry<Filter, FileStatistics>> {
+@Override
+	public int compare(Map.Entry<Filter, FileStatistics> o1, Map.Entry<Filter, FileStatistics> o2) {
```
Надо поддержать равенство тоже:

` The ordering imposed by a comparator c on a set of elements S is said to be consistent with equals if and only if c.compare(e1, e2)==0 has the same boolean value as e1.equals(e2) for every e1 and e2 in S.`

levanovd commented on LoC/src/loc/statistics/StatisticsSerializer.java:
```diff
@@ -0,0 +1,93 @@
+import java.util.*;
+
+public class StatisticsSerializer implements IStatisticsSerializer {
+	public static char space              = ' ';
```
`public static final SPACE`

OkThought committed:
Change design
Add Serializers
Change FilterFactory
Make Serializers and FilterFactory singletons

OkThought committed:
Add checks for null in LoC library entry points

levanovd reviewed (changes requested):
На конфиге
```
  .h  
  
```
выдает
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at loc.filter.FilterFactory.createByString(Unknown Source)
	at loc.filter.FilterFactory.create(Unknown Source)
	at loc.ConfigFileReader.readFilters(Unknown Source)
	at loc.Main.main(Unknown Source)
```

OkThought committed:
HotFix skip empty or whitespace config lines in ConfigFileReader

OkThought committed:
HotFix: log errors on reading file without rights to read and not terminate

OkThought committed:
Add argument checks

OkThought committed:
HotFix: return 0 on equality

OkThought committed:
HotFix: return 0 on strong equality (only when o1.equals(o2))

OkThought committed:
HotFix: make specific fields private and add getters and setters for them

OkThought committed:
Remove commented code

OkThought committed:
Change statistics sorting concept
Use and fill a TreeSet in serialize() method just before main loop

OkThought committed:
Make printVisited private. Add getter and setter for it.
Add one more check for null for filters in constructor

levanovd commented:
```
compile:
    [javac] Compiling 32 source files to /home/levanov/NSU/oop2/15201-bogush/LoC/out/classes
    [javac] /home/levanov/NSU/oop2/15201-bogush/LoC/src/loc/filter/FilterFactory.java:3: error: package com.sun.javaws.exceptions does not exist
    [javac] import com.sun.javaws.exceptions.InvalidArgumentException;
    [javac]                                 ^
    [javac] 1 error
```

OkThought committed:
Remove useless import

RoyPy commented:
Fixed
> On Apr 7, 2017, at 10:13 AM, levanovd <notifications@github.com> wrote:
> 
> compile:
>     [javac] Compiling 32 source files to /home/levanov/NSU/oop2/15201-bogush/LoC/out/classes
>     [javac] /home/levanov/NSU/oop2/15201-bogush/LoC/src/loc/filter/FilterFactory.java:3: error: package com.sun.javaws.exceptions does not exist
>     [javac] import com.sun.javaws.exceptions.InvalidArgumentException;
>     [javac]                                 ^
>     [javac] 1 error
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/RoyPy/15201-bogush/pull/1#issuecomment-292416221>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ANa_ViROOIKkHTv37B6CNn2vJU-6BRekks5rtanHgaJpZM4ML_-l>.
>

levanovd commented:
```
java.lang.ClassCastException: Cannot cast loc.filter.filters.aggregate.Or to loc.filter.filters.aggregate.And
	at java.lang.Class.cast(Class.java:3369)
	at loc.filter.serializers.aggregate.OrFilterSerializer.serialize(Unknown Source)
	at loc.statistics.StatisticsSerializer.serialize(Unknown Source)
	at loc.Main.main(Unknown Source)
```

OkThought committed:
Fix the bug

OkThought committed:
Remove old test

OkThought committed:
Fix prefix

OkThought committed:
Add *FilterSerializer tests

OkThought committed:
Rename to LoCOverallTest

RoyPy commented:
Fixed and added some new tests
> On Apr 7, 2017, at 11:10 AM, levanovd <notifications@github.com> wrote:
> 
> java.lang.ClassCastException: Cannot cast loc.filter.filters.aggregate.Or to loc.filter.filters.aggregate.And
> 	at java.lang.Class.cast(Class.java:3369)
> 	at loc.filter.serializers.aggregate.OrFilterSerializer.serialize(Unknown Source)
> 	at loc.statistics.StatisticsSerializer.serialize(Unknown Source)
> 	at loc.Main.main(Unknown Source)
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/RoyPy/15201-bogush/pull/1#issuecomment-292428905>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ANa_Vqz4qenrwUeKuo3FFdPT2I_eS87gks5rtbdAgaJpZM4ML_-l>.
>

OkThought committed:
Fix indents

OkThought committed:
Remove printVisited + Refactor

levanovd commented:
Почему-то просто .cpp получилось больше, чем ".cpp или что-то еще"
```
Total - 9358226 lines in 19868 files
---------------------
.cpp                  - 11649272 lines in 14804 files
|(.cpp .h .hpp .java) - 9358226  lines in 19868 files
|(.cpp .h .hpp)       - 9308830  lines in 19702 files
|(.cpp .h)            - 9280681  lines in 19537 files
&(.cpp >1400000000)   - 5824636  lines in 7402  files
.h                    - 3456045  lines in 12135 files
.java                 - 49396    lines in 166   files
.hpp                  - 28149    lines in 165   files
```

OkThought committed:
Fix bug with duplicated filters in config
Change ArrayList container to HashSet container

OkThought committed:
Change build dir for jar file

OkThought committed:
Fix cleaning targets

levanovd commented:
Принимается
